### PR TITLE
feat(ext/websocket): websockets over http2

### PR DESCRIPTION
**Pull request by mmastrac**
Implements `WebSocket` over http/2. This requires a conformant http/2 server supporting the extended connect protocol.

Passes approximately 100 new WPT tests (mostly `?wpt_flags=h2` versions of existing websockets APIs).

This is implemented as a fallback when http/1.1 fails, so a server that supports both h1 and h2 WebSockets will still end up on the http/1.1 upgrade path.

The patch also cleas up the websockets handshake to split it up into http, https+http1 and https+http2, making it a little less intertwined.

This uncovered a likely bug in the WPT test server: https://github.com/web-platform-tests/wpt/issues/42896

### CODE REVIEW

File: tools/wpt/expectation.json
Comment (bartlomieju) [inline tools/wpt/expectation.json:1]: Nice!!

File: ext/websocket/stream.rs [hunk 1]
```diff
@@ -109,6 +168,10 @@ impl AsyncWrite for WebSocketStream {
       WsStreamKind::Upgraded(stream) => {
         Pin::new(stream).poll_write_vectored(cx, bufs)
       }
+      WsStreamKind::H2(..) => {
+        // TODO(mmastrac): this is possibly just too difficult, but we'll never call it
+        unimplemented!()
+      }
```
Comment (bartlomieju) [inline ext/websocket/stream.rs:174]: Are we sure this won't be hit?

**Review by bartlomieju — COMMENTED** [review]
Can you add `ci-bench` label and then push an empty commit to make sure everything works correctly?

File: ext/websocket/stream.rs (hunk 1 above)
Comment (mmastrac) [reply ext/websocket/stream.rs:174]: Yep -- I explicitly disabled the writev flag on fastwebsockets for H2.

**Review by bartlomieju — APPROVED** [review]

**Pull request merged**